### ISSUE TITLE
Use microsoft windows-rs crate?

### ISSUE THREAD

**Issue by TomzBench** [rodrigocfd/winsafe#77]
I am curious if there is any intention to migrate some of the lower level stuff like struct definitions and other types to the windows-rs crate? I believe their rust crate is generated and can avoid some potential human errors.

**Comment by rodrigocfd** [comment]
In the early days of this crate, it was a superset of the [winapi](https://crates.io/crates/winapi) crate, which provided all the low-level definitions. At some point, however, I found myself doing a lot of work due to the intrinsic unsafety of those structs.

[LVITEM](https://rodrigocfd.github.io/winsafe/winsafe/struct.LVITEM.html) is a good example: it has a string pointer field, which must **not** be exposed... so I had to write a wrapper to the struct, although I could not hide the old one. And it started to pile up.

So, after walking in circles for a while, it made sense that the low level stuff should be part of the crate itself, what also made it possible to get rid of the heavy dependency.

So, there is **no intention** to go back to that situation.

Unfortunately a safe Win32 wrapper cannot be generated by a computer (so far!), so we have to pay the price of living the possibility of human errors along the process. The fact it is open source is good in this sense – several people have eyed this crate and fixed things.

It will, however, at some point make use of the [windows-targets](https://crates.io/crates/windows-targets) crate, see #75.
